src/test/emulators/auth/signUp.spec.ts

```diff
@@ -395,4 +396,94 @@ describeAuthEmulator("accounts:signUp", ({ authApi }) => {
         expect(res.body.error.message).to.equal("INVALID_PHONE_NUMBER : Invalid format.");
       });
   });
+
+  it("should create new account with multi factor info", async () => {
+    const mfaInfo = { displayName: "Cell Phone", phoneInfo: TEST_PHONE_NUMBER };
+    const user = { email: "alice@example.com", password: "notasecret", mfaInfo: [mfaInfo] };
+    const localId = await registerMfaUser(authApi(), user).then(({ localId }) => {
+      expect(localId).to.be.a("string").and.not.empty;
+      return localId;
+    });
+
+    const info = await getAccountInfoByLocalId(authApi(), localId);
+    expect(info.mfaInfo).to.have.length(1);
+    const savedMfaInfo = info.mfaInfo?.pop();
+    expect(savedMfaInfo).to.include(mfaInfo);
+    expect(savedMfaInfo?.phoneInfo).to.equal(mfaInfo.phoneInfo);
+    expect(savedMfaInfo?.displayName).to.equal(mfaInfo.displayName);
+    expect(savedMfaInfo?.mfaEnrollmentId).to.be.a("string").and.not.empty;
+  });
+
+  it("should create new account with multiple second factors", async () => {
+    const mfaInfo = { displayName: "Cell Phone", phoneInfo: TEST_PHONE_NUMBER };
+    const user = {
+      email: "alice@example.com",
+      password: "notasecret",
+      mfaInfo: [mfaInfo, mfaInfo, mfaInfo],
+    };
+    const localId = await registerMfaUser(authApi(), user).then(({ localId }) => {
+      expect(localId).to.be.a("string").and.not.empty;
+      return localId;
+    });
+
+    const info = await getAccountInfoByLocalId(authApi(), localId);
+    expect(info.mfaInfo).to.have.length(3);
+    for (const savedMfaInfo of info.mfaInfo ?? []) {
+      expect(savedMfaInfo).to.include(mfaInfo);
+      expect(savedMfaInfo?.phoneInfo).to.equal(mfaInfo.phoneInfo);
+      expect(savedMfaInfo?.displayName).to.equal(mfaInfo.displayName);
+      expect(savedMfaInfo?.mfaEnrollmentId).to.be.a("string").and.not.empty;
+    }
+    // each factor should have a unique id...
+    expect(info.mfaInfo![0].mfaEnrollmentId).not.to.eq(info.mfaInfo![1].mfaEnrollmentId);
```

Comment:
I think we can just go with `expect(new Set(info.mfaInfo!.map(info => info.mfaEnrollmentId))).to.have.length(3)`.